File: ria/java/gof/src/main/java/com/gof/pattern/abstractFactory/after/WhiteShipFactory.java
package com.gof.pattern.abstractFactory.after;

import com.gof.pattern.factoryMethod.after.DefaultShipFactory;
import com.gof.pattern.factoryMethod.after.Ship;
import com.gof.pattern.factoryMethod.after.WhiteShip;

public class WhiteShipFactory extends DefaultShipFactory {

    private ShipPartsFactory shipPartsFactory;

    public WhiteShipFactory(ShipPartsFactory shipPartsFactory) {
        this.shipPartsFactory = shipPartsFactory;
    }

    @Override
    public Ship createShip() {
        WhiteShip ship = new WhiteShip();
        ship.setAnchor(shipPartsFactory.createAnchor());
        ship.setWheel(shipPartsFactory.createWheel());
        return ship;
    }
}

File: ria/gof/src/main/kotlin/com/gof/pattern/abstractFactory/after/WhiteShipFactory.kt
package com.gof.pattern.abstractFactory.after

import com.gof.pattern.factory.DefaultShipFactory
import com.gof.pattern.factory.Ship
import com.gof.pattern.factory.WhiteShip

class WhiteShipFactory(private val shipPartsFactory: ShipPartsFactory) : DefaultShipFactory() {

    override fun createShip(): Ship {
        val ship = WhiteShip()
        ship.anchor = shipPartsFactory.createAnchor()
        ship.wheel = shipPartsFactory.createWheel()
        return ship
    }
}
